### PR TITLE
DOC: Typo(s?) in pandas.DataFrame.isin

### LINKED ISSUE #43755: DOC: Typo(s?) in pandas.DataFrame.isin
### 

- [X] I have checked that the issue still exists on the latest versions of the docs on `master` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.isin.html

### Documentation problem

In the last example, the text introducing the code snippet reads:

> When values is a Series or DataFrame the index and column must match. Note that ‘falcon’ does not match based on the number of legs in df2.

However, "df2" does not exist in the  included code snippet (it is instead called "other").

Additionally, I am not comfortable enough with pandas to assert this, but it seems like the text should incorrectly states that "falcon" does not match, since "falcon" indeed does match for both columns, see the relevant portion of the code snippet:

```
>>> df.isin(other)
        num_legs  num_wings
falcon      True       True
dog        False      False
```

### Suggested fix for documentation

Text should at least read:

> When `values` is a Series or DataFrame the index and column must match. Note that ‘falcon’ does not match based on the number of legs in other.

or the variable in the code snippet should be renamed to "df2".

Additionally, something may need to be done to address the potential discrepancy in saying that "falcon" does not match.

### PR DESCRIPTION
**Pull request by Varun270**
- [🗸] closes #43755 
- [🗸] tests added / passed
- [🗸] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [🗸] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -10563,13 +10563,13 @@ def isin(self, values) -> DataFrame:
...
-        in df2.
+        in other.
 
-        >>> other = pd.DataFrame({'num_legs': [8, 2], 'num_wings': [0, 2]},
+        >>> other = pd.DataFrame({'num_legs': [8, 3], 'num_wings': [0, 2]},
         ...                      index=['spider', 'falcon'])
         >>> df.isin(other)
                 num_legs  num_wings
-        falcon      True       True
+        falcon      False       True
```
Comment (jreback) [inline pandas/core/frame.py:10572]: nitpick can you line these up (as would happen in the terminal)

Comment (Varun270) [reply pandas/core/frame.py:10572]: Do you want me to adjust `True` and `False` in line 10572?

Comment (phofl) [reply pandas/core/frame.py:10572]: I think jeff means deleting one whitespace between falcon and false

Comment (Varun270) [reply pandas/core/frame.py]: Understood , I am force pushing the suggested changes

**Comment by Varun270** [comment]
@jreback I did the changes you suggested, you can review them.

**Review by phofl — APPROVED** [review]
Lgtm

**Comment by jreback** [comment]
thanks @Varun270

**Pull request merged**